infra/is-ngcc-necessary.ts

```diff
@@ -0,0 +1,58 @@
+import * as chalk from 'chalk';
+import { existsSync, readFileSync } from 'fs';
+import { resolve } from 'path';
+
+const packages = Object.keys(JSON.parse(readFileSync(resolve(__dirname, '../package.json'), 'utf8')).dependencies);
+const knownNonNgccPackages: string[] = JSON.parse(readFileSync(resolve(__dirname, './non-ngcc-packages.json'), 'utf8'));
+const knownNonNgccPackageMatchers = knownNonNgccPackages.map((pattern: string) => new RegExp(`^${pattern}$`));
```

Comment:
Why use regexes here? It seems like more work than necessary (and - if we wanted to be pedantic - can result in incorrect patterns, because special regex characters are not escaped :grin:).

Wouldn't it be simpler to just use the strings?

Reply:
When I wrote this I expected there to be large numbers of packages that were grouped (e.g. `@anguiar/*`) so I thought this was the best approach.
If we are not going with this, then I would prefer to just list them all explicitly - then, indeed, we could go with a `Set` not that it makes any practical difference to the processing time.